packages/contracts/contracts/rbac.clar

```diff
@@ -0,0 +1,169 @@
+(define-constant CONTRACT_OWNER tx-sender)
+
+;; Stores roles granted to users
+;; Each user can have up to 128 roles ranging from 0-127.
+;; Roles are stored as bitmask values represented as uint's.
+(define-map UserRoles
+  {user: principal}
+  {roles: uint}
+)
+
+;; Stores the roles granted to permissions
+;; Each permission can be granted to up to 128 roles ranging from 0-127
+;; Roles are stored as bitmask values represented as uint's.
+(define-map PermissionRoles
+  {permission: (string-ascii 50)}
+  {roles: uint}
+)
+
+;; Returns all roles granted to a specific user
+;; Returns u0 if user is unknown (has no roles granted)
+(define-read-only (get-roles (user principal)) 
+  (default-to u0 (get roles (map-get? UserRoles {user: user})))
+)
+
+;; Checks if user has a specified role
+(define-read-only (has-role (user principal) (role uint))
+  (begin 
```

Comment:
This could go into a function `contains` and reused in `has-permission`: 

```

(define-read-only (contains (currentRoles uint) (role uint))
  (begin
    (asserts! (> u128 role) false)
    (let
      (
        (searchRole (pow u2 role))
      )
      (> currentRoles (xor currentRoles searchRole))
    )
  )
)
```

Reply:
Yes, yes, yes!! :heart: